packages/node-sdk/README.md

```diff
@@ -0,0 +1,231 @@
+
+// Create a new instance of the client with the secret key. Additional options
+// are available, such as supplying a logging sink, a custom HTTP client and
+// other custom properties.
+//
+// We recommend that only one global instance of `client` should be created
+// to avoid multiple round-trips to our servers.
+const client = new BucketClient({
+  secretKey: "sec_prod_xxxxxxxxxxxxxxxxxxxxx",
+  fallbackFlags: { can_see_new_reports: true },
+});
+
+// Initialize the client and begin fetching flag definitions.
+// You must call this method prior to any calls to `getFlags()`,
+// otherwise an empty object will be returned.
+//
+// You can also supply your `fallbackFlags` to the `initialize()` method. These
+// fallback flags are used in the situation when the server-side flags are not
+// yet loaded or there are issues retrieving them.
+// See "Initialization Options" below for more information.
+await client.initialize();
+```
+
+At any point where the client needs to be used, we can configure it through
+`withUser()`, `withCompany()` and `withOtherContext()` methods. Each of
+these three methods returns a new instance of `Client` class which includes
+the applied user, company or context:
+
+```ts
+// configure the client
+const boundClient = client
+  .withUser("john_doe", { attributes: { name: "John Doe" } })
+  .withCompany("acme_inc", { attributes: { name "Acme Inc."} })
+  .withOtherContext({ additional: "context", number: [1,2,3] })
+
+const fallbackFlags = { can_see_new_reports: true  }
+
+// actively wait for the flags to be loaded
+await boundClient.initialize()
+```
+
+Once the client is initialized, one can obtain the evaluated flags:
+
+```ts
+// get the current flags (uses company, user and custom context to evaluate the flags).
+const flags = boundClient.getFlags();
+
+if (flags.can_see_new_reports) {
+  // this is your flag-protected code ...
+  // send an event when the feature is used:
+  boundClient.trackFeatureUsage("new_reports_used", {
+    attributes: {
+      some: "attribute",
+    },
+  });
+}
+```
+
+## Tracking
+
+Tracking allows sending `user`, `company` and `event` messages to Bucket.
+`user` events can be used to register or update an user's attributes
+with Bucket. `company` allows the same, and additionally, allows
+associating an user with a company on the Bucket side. Finally, `event`
+is used to track feature usage across your application.
+
+The following example shows how to register a new user, and associate it with a company:
+
+```ts
+// registers the user with Bucket using the provided unique ID, and
+// providing a set of custom attributes (can be anything)
+const boundClient = client
+  .withUser("your_user_id", {
+    attributes: { longTimeUser: true, payingCustomer: false },
+  })
+  .withCompany("company_id");
+
+// track the user (send a `user` event to Bucket).
+await boundClient.trackUser();
+
+// register the user as being part of a given company
+boundClient.trackCompany();
```

Comment:
Maybe you and @Munter were already back and forth on this, but did you consider removing the `trackCompany` / `trackUser` methods and instead only relying on `withCompany` / `withUser` ? This is a feature flagging SDK and whenever you do `withCompany` and give it some attributes and then go on to evaluate some flags, those attributes need to be transmitted to our servers in order for the views/tables there to be accurate (who has access to what). So, I think it makes sense for `withCompany` to automatically call `trackCompany` at the least. And then we should ask ourselves if `trackCompany` and `trackUser` is even useful as separate methods.